### DISCUSSION TITLE
What are the limitations of JSqlParser?

### DISCUSSION THREAD

**Discussion by DeepikaEDevadiga** [JSQLParser/JSqlParser#1847] [General]
Is it possible for the JSQL Parser to successfully parse subqueries that are enclosed within multiple sets of parentheses?

**Comment by manticore-projects** [comment]
Yes, especially JSQLParser 4.7 Snapshot supports nested parentheses very well: https://manticore-projects.com/JSQLParser/migration.html#brackets

**Reply by DeepikaEDevadiga** [reply]
Thanks for the reply

**Reply by DeepikaEDevadiga** [reply]
SELECT 1 FROM customers c WHERE EXIST (SELECT 1 FROM orders o JOIN order_items oi ON o.order_id = oi.order_id WHERE c.customer_id = o.customer_id AND oi.product_id IN ( SELECT product_id FROM product_sales WHERE sale_date >= (SELECT MAX(sale_date) FROM sales)));
Can the parser handle the complexity of the given query? I'm working on analyzing the query to identify the tables present along with their corresponding alias names. However, I'm facing difficulties in parsing the query enclosed within parentheses, which is hindering my ability to achieve the desired output.

**Reply by manticore-projects** [reply]
Yes, [you can try it online here.](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20MoUQMiDCAqAECMAoAYgJQPIFlYGMCuAzgC4D2AtgKYBOBuiA6gBIioiwgAeAlsQBSygIMBCgzYSVACbVaJRLAWwAUugCSAOVgTpVAPpciFMrK7zF59JpIA6bdX2TYAXi1dbU+10kNmrXNfxick9HFxtA0ko9LzMFAEF1ABFXawAHKhJJPBwiB1gNWH4BcCg4dMzs3JjzGtq62rQsWHKsnN0CAEMAGwoCWPqB+qYWNk6e3UkOw1gAPhcihUFS2EwOjn4xigmpilgASn7Bo+OThUbsTdo9-f3EAG4gA)
